SPEC.md

```diff
@@ -98,61 +98,104 @@ A build tool is something which reads one or more input artifacts and writes one
 ## Specifications
 ### Artifact ID
 
-GitBOM uses the git object id (as a blob) for an artifact as its artifact ID.
+A hash function over the []byte of an artifact satisfies the canonical, unique, immutable requirements for an artifact identifier.
+
+### Artifact Identifier Types
+The vast majority of source code artifacts are already indexed by their git object identifiers (gitoids) as git objects of type blob.
+
+For this reason GitBOM uses the gitoid for an artifact as its artifact ID.
```

Comment:
```suggestion
For this reason, GitBOM has chosen to use the "gitoid" of an Artifact as its Artifact Identifier.
```

Reply:
In my suggestion, I've capitalized Artifact and Artifact Identifier as these are proper nouns we are defining. I'm not beholden to this choice, though I think it is good form in a specification to delineate which terms are being defined as domain-specific.

Reply:
Suggestion adopted.